tensilelite/Tensile/KernelWriterAssembly.py

```diff
@@ -11487,9 +11487,20 @@ def functionSuffix(self, kernel):
         self.getLdsSize(kernel), self.agprPool.size(), self.states.doubleVgpr) < 2:
       self.states.overflowedResources = 6
 
-    vgprPerCU = 65536
-    vgprPerThreadPerOccupancy = vgprPerCU // kernel["NumThreads"]
-    numWorkGroupsPerCU = vgprPerThreadPerOccupancy // max(self.vgprPool.size(), self.agprPool.size())
+    if self.states.version[0] is 10:
+      vgprPerCU = 1024
+    elif self.states.version[0] is 11:
+      if self.states.version[2] is 2:
+        vgprPerCU = 1024
+      else:
+        vgprPerCU = 1536
+    elif self.states.version[0] is 12:
+      vgprPerCU = 1536
+    else:   # gfx9 series
+      vgprPerCU = 2048
```

Comment:
consider self.archCaps["ArchAccUnifiedRegs"]

Reply:
Why ArchAccUnifiedRegs? This is the physical VGPR size. We only care about the total VGPR usage. If a WG uses VGPRs more than the physical sizes, the HIP API will return error. This check is to avoid the error.

Reply:
If no ArchAccUnifiedRegs, the actual allocation of accVgpr (or archVgpr) would be equal to max(self.vgprPool.size(),self.agprPool.size()).
For gfx908, if NumThreads is 512 and vgprPool.size() is 256, the new formula would calculate numWorkGroupsPerCU as 1, which wouldn't trigger overflowResources. But the actual case is that it wouldn't have enough vgpr.